internal/attestations/authorization.go

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package attestations
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path"
+
+	"github.com/gittuf/gittuf/internal/gitinterface"
+	"github.com/gittuf/gittuf/internal/third_party/go-git"
+	"github.com/gittuf/gittuf/internal/third_party/go-git/plumbing"
+	ita "github.com/in-toto/attestation/go/v1"
+	sslibdsse "github.com/secure-systems-lab/go-securesystemslib/dsse"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+const (
+	AuthorizationPredicateType = "https://gittuf.dev/authorization/v0.1"
+	digestGitCommitKey         = "gitCommit"
+)
+
+var (
+	ErrInvalidAuthorization  = errors.New("authorization attestation does not match expected details")
+	ErrAuthorizationNotFound = errors.New("requested authorization not found")
+)
+
+// Authorization is a lightweight record of a detached authorization in a gittuf
+// repository. It is meant to be used as a "predicate" in an in-toto
+// attestation.
+type Authorization struct {
+	TargetRef    string `json:"targetRef"`
+	FromTargetID string `json:"fromTargetID"`
+	ToTargetID   string `json:"toTargetID"`
+}
+
+// NewAuthorizationAttestation creates a new authorization for the provided
+// information. The authorization is embedded in an in-toto "statement" and
+// returned with the appropriate "predicate type" set.
+func NewAuthorizationAttestation(targetRef, fromTargetID, toTargetID string) (*ita.Statement, error) {
+	predicate := &Authorization{
+		TargetRef:    targetRef,
+		FromTargetID: fromTargetID,
+		ToTargetID:   toTargetID,
+	}
+
+	predicateBytes, err := json.Marshal(predicate)
+	if err != nil {
+		return nil, err
+	}
+
+	predicateInterface := &map[string]any{}
+	if err := json.Unmarshal(predicateBytes, predicateInterface); err != nil {
+		return nil, err
+	}
+
+	predicateStruct, err := structpb.NewStruct(*predicateInterface)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ita.Statement{
+		Type: ita.StatementTypeUri,
+		Subject: []*ita.ResourceDescriptor{
+			{
+				Digest: map[string]string{digestGitCommitKey: toTargetID},
```

Comment:
```suggestion
				Digest: map[string]string{"sha1": toTargetID},
```

This should map to the algorithm, no? https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md

Reply:
https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md#gitcommit-gittree-gitblob-gittag

The spec supports `git` specific objects as well for an additional hint!